src/state-mutation.js

```diff
@@ -12,14 +12,16 @@ const updateReverseRelationship = (
   return (foreignEntities) => {
     return foreignEntities.update(
       foreignEntities.findIndex(
-        item => item.get('id') === relationship.getIn(['data', 'id'])
+        item => item.get('id') == relationship.getIn(['data', 'id'])
```

Comment:
The reason is that in some APIs (such as the python implementation) the first level IDs are shown as integers, yet the ids present on relationships are casted as strings.

I'm not really sure about how should this be handled, but ideally ids should be of the same data type, so maybe this change could be reverted.